### PR TITLE
add AllocateNetwork and FreeNetwork plugin apis

### PR DESCRIPTION
**Pull request by mavenugo**
https://github.com/docker/docker/pull/27287 brought in the support for swarm-mode network v1 plugins. These 2 APIs are introduced to support remote plugins in swarm-mode.

Also fixed a minor error handling issue in createEndpointPath where the EncodeResponse is called twice incorrectly

Signed-off-by: Madhu Venugopal madhu@docker.com

### CODE REVIEW

File: network/api.go [hunk 1]
```diff
@@ -49,6 +53,36 @@ type CapabilitiesResponse struct {
...
+	Options map[string]string
+}
+
+// FreeNetworkRequest is the request to free allocated network in the manager
+type FreeNetworkRequest struct {
+	// The ID of the network to be freed.
+	NetworkID string
+}
+
+// FreeNetworkResponse is the response to a request for freeing a network.
```
Comment (dave-tucker) [inline network/api.go]: I don't see this used anywhere... Is it necessary?

File: network/api.go [hunk 2]
```diff
@@ -242,6 +305,7 @@ func (h *Handler) initMux() {
 		if err != nil {
 			msg := err.Error()
 			sdk.EncodeResponse(w, NewErrorResponse(msg), msg)
+			return
```
Comment (dave-tucker) [inline network/api.go]: This is a bug, but isn't directly related to your changes.
It might be helpful to note that it was changed in the commit message

**Comment by mavenugo** [comment]
@dave-tucker thanks for the review. Updated.

**Comment by dave-tucker** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Comment by Sochene** [comment]
LGTM

**Pull request merged**